kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/resources/manager/ResourceManager.java

```diff
@@ -158,6 +200,58 @@ public final <T extends HasMetadata> void deleteResource(T... resources) {
         }
     }
 
+    public void deleteResources() {
+        if (!storedResources.containsKey(getTestContext().getDisplayName()) || storedResources.get(getTestContext().getDisplayName()).isEmpty()) {
+            LOGGER.info("In context {} is everything deleted", getTestContext().getDisplayName());
+        }
+        else {
+            LOGGER.info("Deleting all resources for {}", getTestContext().getDisplayName());
+        }
+
+        // if stack is created for specific test suite or test case
+        AtomicInteger numberOfResources = storedResources.get(getTestContext().getDisplayName()) != null
```

Comment:
Why is this atomic involved? It looks like we pop and run all the resource items in the stack, why do we also decrement this atomic and include it in the `while` conditions?